Engine/Graphics/Vis.cpp

```diff
@@ -498,15 +498,7 @@ int Vis::get_object_zbuf_val(Vis_ObjectInfo *info) {
     switch (info->object_type) {
         case VisObjectType_Sprite:
         case VisObjectType_Face: {
-            // return info->sZValue;
-
-            struct {
-                unsigned short object_pid;
-                short depth;
-            } res;
-            res.depth = info->depth;
-            res.object_pid = info->object_pid;
-            return *(int *)&res;
+            return info->sZValue;
```

Comment:
are you sure we do not need to include pid into this?
as far as i remember mouse picking (for example picking items in inventory) relies on getting z value out of zbuffer and getting pid of picked object
might also affect picking Actors (sprite monsters) in the world - please try clicking and right clicking (monster info popup) to confirm that still works

Reply:
The sValue is this:

`union {
        int sZValue;
        struct {
            uint16_t object_pid;
            int16_t depth;
        };`

It was commented out, and then get_object_zbuf_val was doing all that cumbersome casting to recalculate it. I was thinking of instead making a the method return a pid-and-depth struct instead, since that'd make it a lot clearer what's going on when calling it.